kie-server-parent/kie-server-services/kie-server-services-scenario-simulation/src/main/java/org/kie/server/services/scenariosimulation/ScenarioSimulationKieServerExtension.java

```diff
@@ -0,0 +1,143 @@
+    private static final Boolean DMN_DISABLED = Boolean.parseBoolean(System.getProperty(KieServerConstants.KIE_DMN_SERVER_EXT_DISABLED, "false"));
+
+    private KieServerRegistry context;
+    private ScenarioSimulationService scenarioSimulationService;
+    private boolean initialized = false;
+
+    @Override
+    public boolean isInitialized() {
+        return initialized;
+    }
+
+    /**
+     * This extension is active if it is not disabled and one of DMN or Drools is active
+     * @return
+     */
+    @Override
+    public boolean isActive() {
+        return !SCESIM_DISABLED && (!DROOLS_DISABLED || !DMN_DISABLED);
+    }
+
+    @Override
+    public void init(KieServerImpl kieServer, KieServerRegistry registry) {
+        this.context = registry;
+        this.scenarioSimulationService = new ScenarioSimulationService(context);
+        initialized = true;
+    }
+
+    @Override
+    public void destroy(KieServerImpl kieServer, KieServerRegistry registry) {
+        //no-op
+    }
+
+    @Override
+    public void createContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        //no-op
+    }
+
+    @Override
+    public boolean isUpdateContainerAllowed(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        return true;
+    }
+
+    @Override
+    public void updateContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        //no-op
+    }
+
+    @Override
+    public void disposeContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        //no-op
+    }
+
+    @Override
+    public List<Object> getAppComponents(SupportedTransports type) {
+        ServiceLoader<KieServerApplicationComponentsService> appComponentsServices
+                = ServiceLoader.load(KieServerApplicationComponentsService.class);
+        List<Object> appComponentsList = new ArrayList<>();
+        Object[] services = {context, scenarioSimulationService};
+        for (KieServerApplicationComponentsService appComponentsService : appComponentsServices) {
+            appComponentsList.addAll(appComponentsService.getAppComponents(EXTENSION_NAME, type, services));
+        }
+
+        return appComponentsList;
+    }
+
+    @Override
+    public <T> T getAppComponents(Class<T> serviceType) {
+        return null;
+    }
+
+    @Override
+    public String getImplementedCapability() {
+        return KieServerConstants.CAPABILITY_SCENARIO_SIMULATION;
+    }
+
+    @Override
+    public List<Object> getServices() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public String getExtensionName() {
+        return EXTENSION_NAME;
+    }
+
+    @Override
+    public Integer getStartOrder() {
+        return Integer.MAX_VALUE;
```

Comment:
What about using some lower number here, to allow possible other extensions to depend on this extension?

Reply:
This extension is not designed to be used by any other extensions so I prefer to keep this value. If this will be needed in the future we can change but I really don't think will happen.

Reply:
ok